src/pynwb/file.py

```diff
@@ -275,7 +275,10 @@ class NWBFile(MultiContainerInterface):
             {'name': 'electrode_groups', 'type': Iterable,
              'doc': 'the ElectrodeGroups that belong to this NWBFile', 'default': None},
             {'name': 'ic_electrodes', 'type': (list, tuple),
-             'doc': 'IntracellularElectrodes that belong to this NWBFile', 'default': None},
+             'doc': 'DEPRECATED use icephys_electrodes parameter instead. '
+                    'IntracellularElectrodes that belong to this NWBFile', 'default': None},
+            {'name': 'icephys_electrodes', 'type': (list, tuple),
```

Comment:
Until we enforce the use of keyword-only arguments in neurodata_type constructors, the new docval argument should be added to the end of the list to maintain backward compatibility. That said, I cannot imagine anyone specifying positional arguments through this point in the `NWBFile` argument list, so adding the argument here is almost certainly fine.

Reply:
Thanks for the suggestion @rly! I updated the code accordingly.